source/code/include/playfab/PlayFabAuthenticationContext.h

```diff
@@ -11,16 +11,20 @@ namespace PlayFab
     {
     public:
 #ifndef DISABLE_PLAYFABCLIENT_API
+        // DisableAdvertising is provided for completeness, but changing it is not suggested
+        // Disabling this may prevent your advertising-related PlayFab marketplace partners from working correctly
+        bool disableAdvertising;
+        std::string playFabId; // Master_Player_Entity Id for the Player that logged in
         std::string clientSessionTicket; // Client session ticket that is used as an authentication token in many PlayFab API methods.
+        std::string advertisingIdType; // Set this to the appropriate AD_TYPE_X constant below
+        std::string advertisingIdValue; // Set this to corresponding device value
 #endif
-#ifndef DISABLE_PLAYFABENTITY_API
+        std::string entityId; // Entity Id for the active entity
+        std::string entityType; // Entity Type for the active entity
         std::string entityToken; // User's entity token. Entity tokens are required by all Entity API methods.
-#endif
-#if defined(ENABLE_PLAYFABSERVER_API) || defined(ENABLE_PLAYFABADMIN_API)
-        std::string developerSecretKey; // Developer secret key. These keys can be used in development environments.
-#endif
 
         PlayFabAuthenticationContext();
+        void HandlePlayFabLogin(const std::string& setPlayFabId, const std::string& setClientSessionTicket, const std::string& setEntityId, const std::string& setEntityType, const std::string& setEntityToken);
```

Comment:
>set [](start = 51, length = 3)

Why are these named as with "set" prefix ? Is that a convention we're following in other parts of the code ?

Reply:
I did it to avoid warnings about member variables and function variables having the same name.
(I have my warning level set pretty high)
Do we do it elsewhere in XPlat - maybe, but probably not.
Do we do it elsewhere in other SDKs - yes.

---
In reply to: [343282558](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343282558) [](ancestors = 343282558)

Reply:
We should figure out how to adhere to the same convention

---
In reply to: [343315277](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343315277) [](ancestors = 343315277,343282558)